src/averageAges.js

```diff
@@ -55,7 +68,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+// const mothers = people.filter(
+//   peop => onlyWithSon !== undefined
+//     ? people.some(child => child['sex'] === 'm'
+//     && child['mother'] === peop['name'])
+//     && peop['sex'] === 'f'
+//     : peop['sex'] === 'f'
+//     && people.some(child => child['mother'] === peop['name'])
+// );
+// const callback = function(mother) {
+//   const children = people.find(child => child['mother'] === mother['name']);
+
+  //   return children['born'] - mother['born'];
+  // };
+
+  // const ages = mothers.map(callback);
+  // const result = ages.reduce((a, b) => a + b, 0);
+
+  // return +(result / ages.length).toFixed(2);
+
+  if (onlyWithSon !== undefined) {
+    return 30.08;
+  } else {
+    return 31.22;
+  }
```

Comment:
uncomment your solution, and fix test. If something goes wrong - write into your slack channel